config/src/client.rs

```diff
@@ -124,20 +124,30 @@ impl ConfigurationProxy {
         if let Some(tx_ttl) = self.transaction_time_to_live_ms {
             // Really small TTL would be detrimental to performance
             if tx_ttl < TTL_TOO_SMALL_THRESHOLD {
-                eyre::bail!(
-                    ConfigError::ProxyBuildError("`TRANSACTION_TIME_TO_LIVE_MS`, network throughput may be compromised for values less than {TTL_TOO_SMALL_THRESHOLD}".to_owned())
-                );
+                eyre::bail!(ConfigError::InsaneValue {
+                    field: "TRANSACTION_TIME_TO_LIVE_MS",
```

Comment:
I believe that hard-coding SCREAMING field name should be replaced with _at least_ a `const`:

```rust
struct ConfigFieldsScreaming;

impl ConfigFieldsScreaming {
  pub const TxTimeToLiveMs: &'static str = "TRANSACTION_TIME_TO_LIVE_MS";
}

ConfigError::InsaneValue { field: ConfigFieldsScreaming::TxTimeToLiveMs }
```

Reply:
In theory, yes. In practice, 1. Rustc would complain that your constant isn't upper-snake-case (screaming), and 2. The Rust-y name for the constant is going to be precisely the same as its string representation. 

Best practices from other languages would suggest that naming a constant is a good idea, citing refactoring ability. These don't apply here for the following reasons: 
1. The symbol name is straightforward to find/replace, and it would also affect docs (which is good). 
2. And if you did change the string, you probably also need to change the variable name, because it is misleading. 
3. We don't want the config parameters to change names often if at all. Most headaches that @BAStos525 @ilchu and @Mingela are facing are because we change this too frequently.